docusaurus/docs/reference/ui-extensions.md

```diff
@@ -0,0 +1,407 @@
+---
+id: ui-extensions
+title: UI extensions
+description: Reference for UI extensions
+keywords:
+  - grafana
+  - plugins
+  - documentation
+  - plugin.json
+  - API reference
+  - UI extensions
+sidebar_position: 50
+---
+
+This page describes the UI Extensions API in detail.
+
+## Extension points in Grafana
+
+A list of available extension points within Grafana that can be extended by plugins. All these extension point IDs can be accessed using the [`PluginExtensionPoints` enum exposed by the `@grafana/data`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L121) package.
+
+```typescript
+import { PluginExtensionPoints } from '@grafana/data';
+
+const extensionPointId = PluginExtensionPoints.DashboardPanelMenu;
+```
+
+| Property                          | Type      | Description                                                          |
+| --------------------------------- | --------- | -------------------------------------------------------------------- |
+| **`AlertingAlertingRuleAction`**  | Link      | Extend the alert rule menu with custom actions.                      |
+| **`AlertingHomePage`**            | Component | Extend the alerting home page with custom alert-creation experience. |
+| **`AlertingRecordingRuleAction`** | Link      | Extend the alert rule menu with custom actions.                      |
```

Comment:
This is duplicate of Alerting rule, just to check both come under alert rule menu?

Reply:
Seems like it https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/plugins/useRulePluginLinkExtensions.ts/#L64-L87. But would be good if @mckn could confirm

Reply:
I think it depends on what kind of alert rule you currently view. There are two major types recording rules and alert rules. I think we should add that to the description of the extension point.